textattack/transformations/word_swap_hownet.py

```diff
@@ -0,0 +1,103 @@
+import pickle
+
+from flair.data import Sentence
+from flair.models import SequenceTagger
+
+from textattack.shared import utils
+from textattack.transformations.word_swap import WordSwap
+
+
+class WordSwapHowNet(WordSwap):
+    """ Transforms an input by replacing its words with synonyms in the stored synonyms bank
+    generated by the OpenHowNet. """
+
+    PATH = "hownet"
+
+    def __init__(self, max_candidates=-1, **kwargs):
+        super().__init__(**kwargs)
+        self.max_candidates = max_candidates
+
+        # Download embeddings if they're not cached.
+        cache_path = utils.download_if_needed(
+            "{}/{}".format(WordSwapHowNet.PATH, "word_candidates_sense.pkl")
+        )
+
+        # Actually load the files from disk.
+        with open(cache_path, "rb") as fp:
+            self.candidates_bank = pickle.load(fp)
+
+        self._flair_pos_tagger = SequenceTagger.load("pos-fast")
+        self.pos_dict = {"JJ": "adj", "NN": "noun", "RB": "adv", "VB": "verb"}
+
+    def _get_replacement_words(self, word, word_pos):
+        """ Returns a list of possible 'candidate words' to replace a word in a sentence 
+            or phrase. Based on nearest neighbors selected word embeddings.
+        """
+        word_pos = self.pos_dict.get(word_pos, None)
+        if word_pos is None:
+            return []
+
+        try:
+            candidate_words = self.candidates_bank[word.lower()][word_pos]
+            if self.max_candidates > 0:
+                candidate_words = candidate_words[: self.max_candidates]
+            return [
+                recover_word_case(candidate_word, word)
+                for candidate_word in candidate_words
+            ]
+        except KeyError:
+            # This word is not in our synonym bank, so return an empty list.
+            return []
+
+    def _get_transformations(self, current_text, indices_to_modify):
+        words = current_text.words
+        words_str = " ".join(words)
+        word_list, pos_list = zip_flair_result(
+            self._flair_pos_tagger.predict(words_str)[0]
```

Comment:
`words_str` seems to just be `current_text.text` with the punctuation removed

why not `zip_flair_result(self._flair_pos_tagger.predict(current_text.text)[0])`?

Reply:
since I need to get the POS tag for each word in the "words" list, and from the text to words list, many non-alphabet things have been removed, I do not know map from the POS results of text to each word in the words list.